pandas/core/ops/blockwise.py

```diff
@@ -0,0 +1,91 @@
+from typing import TYPE_CHECKING, List, Tuple
+
+import numpy as np
+
+from pandas._typing import ArrayLike
+
+if TYPE_CHECKING:
+    from pandas.core.internals.blocks import Block  # noqa:F401
+
+
+def operate_blockwise(left, right, array_op):
+    assert right._indexed_same(left)
+
+    def get_same_shape_values(
```

Comment:
this should be moved out to a function in the module; this is complex (partialy) because of this

Reply:
updated